### ISSUE TITLE
训练自己数据的问题

### ISSUE THREAD

**Issue by LauryMiao** [chineseocr/chinese-ocr#46]
1. 请问数据量多大比较好？ 我目前用SynthText_Chinese_version 生成了1000+数据，形式如下：
![ex_1](https://user-images.githubusercontent.com/13173000/34597076-828491b2-f21f-11e7-8df1-64bc964af861.jpg)
[ex_1.txt](https://github.com/chineseocr/chinese-ocr/files/1605651/ex_1.txt)
![ex_2](https://user-images.githubusercontent.com/13173000/34597077-82db5218-f21f-11e7-8bde-e2f6b9c8e7c8.jpg)
[ex_2.txt](https://github.com/chineseocr/chinese-ocr/files/1605652/ex_2.txt)
![ex_3](https://user-images.githubusercontent.com/13173000/34597078-833a4714-f21f-11e7-8a71-8b8ecbc2dae7.jpg)
[ex_3.txt](https://github.com/chineseocr/chinese-ocr/files/1605653/ex_3.txt)
![ex_4](https://user-images.githubusercontent.com/13173000/34597079-8455f5a8-f21f-11e7-9a44-fecdae5c8a8f.jpg)
[ex_4.txt](https://github.com/chineseocr/chinese-ocr/files/1605654/ex_4.txt)
![ex_5](https://user-images.githubusercontent.com/13173000/34597080-84b26f18-f21f-11e7-9d5f-7f0575e28c48.jpg)
[ex_5.txt](https://github.com/chineseocr/chinese-ocr/files/1605655/ex_5.txt)
![ex_6](https://user-images.githubusercontent.com/13173000/34597081-850aa17e-f21f-11e7-8e36-7e30a74072d5.jpg)
[ex_6.txt](https://github.com/chineseocr/chinese-ocr/files/1605656/ex_6.txt)
但是用keras训练出来的效果并不好，loss一直在100多，然后将生成的h5替换了ocr下面的h5，做demo的时候就特别差，比repo自带的h5还差。我看到creat_dataset.py里`label = labelList[i]`， 我理解的是直接将label(txt)的地址写到db里，对吗？然后结果差的原因是否是因为我生成样本用的字体和真正检测时的字体不同生成的？

2. 用repo里自带的H5, 测出的效果有乱码，请问这种应该怎么处理比较好？
![11](https://user-images.githubusercontent.com/13173000/34597199-4346e634-f220-11e7-9594-d1c2b0666d33.png)
![22](https://user-images.githubusercontent.com/13173000/34597200-4372603e-f220-11e7-9cd7-59dff431cea6.png)
![33](https://user-images.githubusercontent.com/13173000/34597201-439d2ca6-f220-11e7-9d4a-eeab0caca839.png)
![44](https://user-images.githubusercontent.com/13173000/34597202-43ca8f2a-f220-11e7-9cc4-3db19417e1a2.png)

```
 着值4单位
_检验项目
结果
J码
It takes time:1.84905695915s
---------------------------------------
红细胞:
 红蛋白含
红细胞!
浓度
红蛋
红细胞分布宽胃
It takes time:0.739130973816s
---------------------------------------
_代码
检验项巨
参考值/总位
结果
It takes time:1.89479899406s
---------------------------------------
单核细胞绝对值
嗜酸性细胞绝对值
嗜碱性粒细胞绝对值
```

对大图的效果如下：
![66](https://user-images.githubusercontent.com/13173000/34597267-a75b778e-f220-11e7-9199-49b84409fcf2.png)

pytorch
```
室查宣上堂位
_绝基
 参 值4位_上  _趁些贸旦_
 缠墨
_代 _检验烫旦_
 0.22-l.70mol/
甘油三函
-1.16
3.4-20.5umol/LHG
 16.9
总胆红素
.TBIL
2.33-5.60mmoL/
4.60
总胆固醇
o0-6.8umol/LACHOL
6.0
直接胆红素
DBIL
0.92-l.84mmol/
3.4-13.0umol/LHDL-C高密度脂蛋白胆固醇.79
 10.90
司接胆红素
IBIL
 3.12mmol/L
0.2-0.4iDL-C密度脂蛋白胆固醇2.40
0.36
意直胆/总胆
DB/T
7-40U/L
14
谷丙转贫醇
ALT
13-40U/Ll
15
谷草转氨酶
AS7
-1.07
AST/A谷草/谷丙
-3.9-6.1mmol/Ll
4.6
葡萄糖
GLU
90-420umol/L
367
尿酸
JA
22.5-8.2mmol/Ll
5.0
尿素氨
BURN
44-106umol/Ll
85
CREA肌哥
0-0.1
0,06
BLM/CR尿素氨/肌哥
70-205l/min/1
91.7
3FR内生肌醉清除率

```
keras
```
室查宣上堂位
_绝基
 参 值4位_上  _趁些贸旦_
 缠墨
_代 _检验烫旦_
 0.22-l.70mol/
甘油三函
-1.16
3.4-20.5umol/LHG
 16.9
总胆红素
.TBIL
2.33-5.60mmoL/
4.60
总胆固醇
o0-6.8umol/LACHOL
6.0
直接胆红素
DBIL
0.92-l.84mmol/
3.4-13.0umol/LHDL-C高密度脂蛋白胆固醇.79
 10.90
司接胆红素
IBIL
 3.12mmol/L
0.2-0.4iDL-C密度脂蛋白胆固醇2.40
0.36
意直胆/总胆
DB/T
7-40U/L
14
谷丙转贫醇
ALT
13-40U/Ll
15
谷草转氨酶
AS7
-1.07
AST/A谷草/谷丙
-3.9-6.1mmol/Ll
4.6
葡萄糖
GLU
90-420umol/L
367
尿酸
JA
22.5-8.2mmol/Ll
5.0
尿素氨
BURN
44-106umol/Ll
85
CREA肌哥
0-0.1
0,06
BLM/CR尿素氨/肌哥
70-205l/min/1
91.7
3FR内生肌醉清除率
```

请问如果想减小乱码的机率，需要怎样处理？

**Comment by wenlihaoyu** [comment]
没有看懂你想表述的是什么？是文本检测不好还是OCR效果不好？

**Comment by LauryMiao** [comment]
不好意思啊，没描述清楚。

简单说就是用你的repo的h5，效果不错，但还是有部分乱码；所以我想继续训练，但结果比你的H5结果还差。我主要想做这种检验报告的检测（图片到文字）。

所以想请教的是：1. 是不是我的训练集做得不好？ 2. 如果抛开我的想法，您这边有什么建议来提高原来repo识别正确的概率（针对体检报告，我网上搜的体检报告）

**Comment by axzhcode** [comment]
同样的问题。
针对特定的图片进行训练生成的结果，并没有原来的的model结果好。。。
不知道最终生成的model用了多少数据？

**Comment by LiangHao92** [comment]
@Paul0M 你好，请问你用SynthText_Chinese_version生成数据限定了字符个数么？不限定字符个数也能正常训练么

**Comment by chengchaoyang** [comment]
@Paul0M @axzhcode 请问你们的问题解决了吗，我也是用自己生成的数据训练以后，还没有博主之前的模型识别效果好，同问为什么。是因为训练数据太少了吗，还是因为其他的？不知道博主训练用了什么样的数据？

**Comment by Nghingtim** [comment]
@chengchaoyang你好，请问在训练自己的模型时，数据长度有限制吗？然后训练出来的识别准备率是多少呢？
